src/google/cloud/ndb/query.py

```diff
@@ -973,28 +1009,29 @@ def __init__(
                     "filters must be a query Node or None; "
                     "received {}".format(filters)
                 )
-        if orders is not None:
-            if not isinstance(orders, (list,)):  # datastore_query.Order
+        if order_by is not None:
+            if not isinstance(order_by, (list, tuple)):
                 raise TypeError(
-                    "orders must be an Order instance or None; "
-                    "received {}".format(orders)
+                    "order must be a list, a tuple or None; "
+                    "received {}".format(order_by)
                 )
-        # if default_options is not None:  # Optional QueryOptions object.
-        #     if not isinstance(default_options, datastore_rpc.BaseConfiguration):
-        #         raise TypeError("default_options must be a Configuration or None; "
-        #                         "received {}".format(default_options))
-        #     if projection is not None:
-        #         if default_options.projection is not None:
-        #             raise TypeError("cannot use projection keyword argument and "
-        #                             "default_options.projection at the same time")
-        #         if default_options.keys_only is not None:
-        #             raise TypeError("cannot use projection keyword argument and "
-        #                             "default_options.keys_only at the same time")
+        if default_options is not None:
+            if not isinstance(default_options, QueryOptions):
+                raise TypeError(
+                    "default_options must be QueryOptions or None; "
+                    "received {}".format(default_options)
+                )
+            if projection is not None:
+                if getattr(default_options, "projection", None) is not None:
+                    raise TypeError(
+                        "cannot use projection keyword argument and "
+                        "default_options.projection at the same time"
+                    )
 
         self.kind = kind
         self.ancestor = ancestor
         self.filters = filters
-        self.orders = orders
+        self.order_by = self.orders = order_by
```

Comment:
Ah, I see. You kept `orders` as an attribute. I think the signature is what we need to preserve for backwards compatibility. We can store it as just `order` on the object.